okhttp/src/main/java/okhttp3/internal/http/RealInterceptorChain.kt

```diff
@@ -84,7 +84,7 @@ class RealInterceptorChain(
     calls++
 
     if (exchange != null) {
-      check(exchange.connection.supportsUrl(request.url)) {
+      check(exchange.finder.sameHostAndPort(request.url)) {
```

Comment:
This is a similar bug fix. If a network interceptor changes the URL from a.example.com to b.example.com, we want to explode even if the connection is actually capable of that request.